core/src/main/java/org/infinispan/persistence/PersistenceUtil.java

```diff
@@ -37,26 +38,30 @@ public static KeyFilter notNull(KeyFilter filter) {
       return filter == null ? KeyFilter.ACCEPT_ALL_FILTER : filter;
    }
 
-   public static <K, V> int count(AdvancedCacheLoader<K, V> acl, KeyFilter<? super K> filter) {
-      final AtomicInteger result = new AtomicInteger(0);
-      acl.process(filter, (marshalledEntry, taskContext) -> result.incrementAndGet(), new WithinThreadExecutor(), false, false);
-      return result.get();
+   public static <K, V> int count(AdvancedCacheLoader<K, V> acl, Predicate<? super K> filter) {
```

Comment:
This class is public, you need to make the change backwards-compatible.

Reply:
above will fix that